docs/standard/native-interop/type-marshalling.md

```diff
@@ -1,18 +1,21 @@
 ---
 title: Type marshalling - .NET
 description: Learn how .NET marshals your types to a native representation.
-ms.date: 01/18/2019
+ms.date: 04/08/2024
 ---
 
 # Type marshalling
 
 **Marshalling** is the process of transforming types when they need to cross between managed and native code.
 
-Marshalling is needed because the types in the managed and unmanaged code are different. In managed code, for instance, you have a `String`, while in the unmanaged world strings can be Unicode ("wide"), non-Unicode, null-terminated, ASCII, etc. By default, the P/Invoke subsystem tries to do the right thing based on the default behavior, described on this article. However, for those situations where you need extra control, you can employ the [MarshalAs](xref:System.Runtime.InteropServices.MarshalAsAttribute) attribute to specify what is the expected type on the unmanaged side. For instance, if you want the string to be sent as a null-terminated ANSI string, you could do it like this:
+Marshalling is needed because the types in the managed and unmanaged code are different. In managed code, for instance, you have a `string`, while unmanaged strings can be .NET Unicode (UTF-16), ANSI Code Page, UTF-8, null-terminated, ASCII, etc. By default, the P/Invoke subsystem tries to do the right thing based on the default behavior, described in this article. However, for those situations where you need extra control, you can employ the [MarshalAs](xref:System.Runtime.InteropServices.MarshalAsAttribute) attribute to specify what is the expected type on the unmanaged side. For instance, if you want the string to be sent as a null-terminated UTF-8 string, you could do it like this:
```

Comment:
I've never see the term ".NET Unicode". I had this view that a .NET string passed out to native code would be seen as a regular UTF16 string. Is that incorrect?

Reply:
Agreed. It is not great. I kept it because of [`CharSet.Unicode`](https://learn.microsoft.com/dotnet/api/system.runtime.interopservices.charset). Open to suggestions. Do you think it is clearer to say ".NET's natural encoding (UTF-16)"?

Reply:
I think of it as ".NET `string` encoding (UTF-16)". I think most people should grasp what that means. I realize that `string` is a C# ism but I think it is more clear than not.